rpc/internal/perf_service.go

```diff
@@ -96,8 +83,8 @@ func (srv *perfService) AttachArtifacts(ctx context.Context, artifactData *Artif
 	resp := &MetricsResponse{}
 	resp.Id = record.ID
 
-	if err := srv.addArtifacts(ctx, record, artifactData.Artifacts); err != nil {
-		return resp, err
+	for _, a := range artifactData.Artifacts {
```

Comment:
It's fine for AttachArtifacts to not send updates now, since it will all be sent together in CloseMetrics. Right?

Reply:
Exactly! We only want to do this once to prevent duplicating the work here.